### DISCUSSION TITLE
Search supports multiple languages

### DISCUSSION THREAD

**Discussion by zjy365** [colinwilson/lotusdocs#84] [Feature Requests]
Search supports multiple languages
When I was using flexsearch, I wanted to set up Chinese search, and the content of the document was in Chinese, but it couldn't be searched.
I tried a lot of things, referring to these links
- https://github.com/alex-shpak/hugo-book/issues/80
- https://github.com/thetre97/gridsome-plugin-flexsearch/issues/23
- https://github.com/thetre97/gridsome-plugin-flexsearch/issues/23#issuecomment-604475110
- https://github.com/nextapps-de/flexsearch#cjk-word-break-chinese-japanese-korean

**Comment by colinwilson** [comment]
Should be a way to incorporate this as an option(s) in the theme config.

**Comment by zjy365** [comment]
@colinwilson 
I tried to modify this part of the code in flexsearch.html, but it didn't work

```
    var index = new FlexSearch.Document({
        // charset: "latin:default",
        tokenize: {{ .Site.Params.flexsearch.tokenize | default "forward" }},
        minlength: {{ .Site.Params.flexsearch.minQueryChar | default 0}},
        cache: {{ .Site.Params.flexsearch.cache | default 100 }},
        optimize: {{ .Site.Params.flexsearch.optimize | default true }},
        document: {
        id: 'id',
        store: [
            "href", "title", "description"
        ],
        index: ["title", "description", "content"]
        }
    });
```

I added this code
```
  encode: false,
  tokenize: function(str){
    return str.replace(/[\x00-\x7F]/g, "").split("");
  }
```

**Comment by colinwilson** [comment]
To be honest, this is what I was intending to try.

I'll have a closer look at this when I have more time over the coming weekend.

**Reply by yangchuansheng** [reply]
any progress?
